design/continuous-delivery.md

```diff
@@ -0,0 +1,332 @@
+# Continuous Delivery for CDK Apps
+
+This is a specification for continuous delivery support for CDK apps. It
+describes the requirements, proposed APIs and developer workflow.
+
+**_Goal_: full CI/CD for CDK apps at any complexity.**
+
+The desired developer experience is that teams will be able to “git push” a
+change into their CDK app repo and have this change automatically picked up,
+built, tested and deployed according to a deployment flow they define.
+
+Any changes in resources, assets *or stacks* in their apps will automatically be
+added. To that end, the deployment pipeline itself will also be continuously
```

Comment:
How will those roles required for the deployment of the pipeline work? will they short lived roles that are created and destroyed?

Reply:
No, they will be created during the environment bootstrapping process.